lib/ipsecbase.pm

```diff
@@ -0,0 +1,120 @@
+    assert_script_run("ip -6 addr add $args{ip} dev $args{dev}");
+}
+
+sub add_ipv6_route {
+    my ($self, %args) = @_;
+    assert_script_run("ip -6 route add $args{dst} via $args{via}");
+}
+
+sub upload_service_log {
+    my ($self, $service_name, $args) = @_;
+    script_run("journalctl -u $service_name -o short-precise > /tmp/$service_name");
+    script_run("cat /tmp/$service_name");
+    upload_logs("/tmp/$service_name", failok => 1, log_name => $args->{log_name});
+}
+
+sub destroy_test_barriers {
+    my ($self) = @_;
+    barrier_destroy('IPSEC_IP_SETUP_DONE');
+    barrier_destroy('IPSEC_ROUTE_SETUP_DONE');
+    barrier_destroy('IPSEC_TUNNEL_MODE_SETUP_DONE');
+    barrier_destroy('IPSEC_SET_MTU_DONE');
+    barrier_destroy('IPSEC_TUNNEL_MODE_CHECK_DONE');
+    barrier_destroy('IPSEC_TRANSPORT_MODE_SETUP_DONE');
+    barrier_destroy('IPSEC_TRANSPORT_MODE_CHECK_DONE');
+}
+
+sub post_run_hook {
+    my ($self) = @_;
+    select_console('log-console');
+    my $hname = get_var('HOSTNAME', 'susetest');
+    $self->upload_service_log("wicked");
+    upload_logs('/var/log/zypper.log');
+}
+
+sub post_fail_hook {
+    my ($self) = @_;
+    #$self->destroy_test_barriers();
+    select_serial_terminal;
+    script_run("SUSEConnect --status-text");
+    script_run("journalctl -o short-precise > /tmp/journal.log");
+    script_run('cat /tmp/journal.log');
+    upload_logs('/tmp/journal.log', failok => 1);
+    upload_service_log('wickedd-dhcp4.service');
+}
+
+sub check_ipv6_addr {
+    my $errors = 0;
+    my $tries = 10;
+    my $no_ip = 1;
+    my $output = '';
+    while ($tries > 0 && $no_ip) {
+        $no_ip = 0;
+        $output = script_output('ip a');
+        if ($output =~ /tentative/) {
+            record_info('Waiting for IPv6 ready, still tentative state');
+            $no_ip = 1;
+        }
+        $tries -= 1;
+        sleep(5);
+    }
+}
+
+sub config_ipsec {
+    my ($self, $args) = @_;
+
+    assert_script_run("ip xfrm state flush &&  ip xfrm policy flush");
+    assert_script_run("ip xfrm state add src $args->{local_ip} dst $args->{remote_ip} proto $self->{ipsec_proto} spi $self->{ipsec_id} reqid $self->{ipsec_id} mode $self->{ipsec_mode} aead $self->{ipsec_aead} $self->{ipsec_key} 128");
+    assert_script_run("ip xfrm state add src $args->{remote_ip} dst $args->{local_ip} proto $self->{ipsec_proto} spi $self->{ipsec_id} reqid $self->{ipsec_id} mode $self->{ipsec_mode} aead $self->{ipsec_aead} $self->{ipsec_key} 128");
+    assert_script_run("ip xfrm policy add src $args->{new_local_net} dst $args->{new_remote_net} dir out tmpl src $args->{local_ip} dst $args->{remote_ip} proto $self->{ipsec_proto} reqid $self->{ipsec_id} mode $self->{ipsec_mode}");
+    assert_script_run("ip xfrm policy add src $args->{new_remote_net} dst $args->{new_local_net} dir in tmpl src $args->{remote_ip} dst $args->{local_ip} proto $self->{ipsec_proto} reqid $self->{ipsec_id} mode $self->{ipsec_mode}");
+}
+
+sub prepare_env {
```

Comment:
This function is used only once. Are there plans to use it in the future?

Reply:
Not sure about it.

Reply:
Will put into pre_run hook.